azresources/analytics/synapse/synapse-without-cmk.bicep

```diff
@@ -126,28 +144,42 @@ resource synapse 'Microsoft.Synapse/workspaces@2021-03-01' = {
       }
     }
   }
+}
 
-  resource synapse_audit 'auditingSettings@2021-05-01' = {
-    name: 'default'
-    properties: {
-      isAzureMonitorTargetEnabled: true
-      state: 'Enabled'
-    }
+
+resource synapse_audit 'Microsoft.Synapse/workspaces/auditingSettings@2021-05-01' = {
```

Comment:
You need a dependency on the parent `synapse` resource. Otherwise, ARM will not wait for synapse to be deployed.

```
{
    "status": "Failed",
    "error": {
        "code": "ResourceNotFound",
        "message": "The Resource 'Microsoft.Synapse/workspaces/syn52iwf6moxedg2' under resource group 'health2022Q1Compute' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
    }
}
```